src/streamingCalls/streaming.ts

```diff
@@ -113,10 +127,192 @@ export class StreamProxy extends duplexify implements GRPCCallResult {
+      error.code = Status.DEADLINE_EXCEEDED;
+      this.emit('error', error);
+
+      this.destroy(error);
+      // Without throwing error you get unhandled error since we are returning a new stream
+      // There might be a better way to do this
+      throw error;
+    }
+
+    if (this.retries && this.retries > maxRetries) {
+      const error = new GoogleError(
+        'Exceeded maximum number of retries before any ' +
+          'response was received'
+      );
+      error.code = Status.DEADLINE_EXCEEDED;
+      this.emit('error', error);
+
+      this.destroy(error);
+      throw error;
+    }
+  }
+
+  /**
+   * Error handler for server streaming retries
+   * @param {CancellableStream} stream - the stream being retried
+   * @param {RetryOptions} retry - Configures the exceptions upon which the
+   *   function should retry, and the parameters to the exponential backoff retry
+   *   algorithm.
+   * @param {Error} error - error to handle
+   */
+  streamHandoffErrorHandler(
+    stream: CancellableStream,
+    retry: RetryOptions,
+    error: Error
+  ): void {
+    let retryStream = this.stream;
+    const delayMult = retry.backoffSettings.retryDelayMultiplier;
+    const maxDelay = retry.backoffSettings.maxRetryDelayMillis;
+    const timeoutMult = retry.backoffSettings.rpcTimeoutMultiplier;
+    const maxTimeout = retry.backoffSettings.maxRpcTimeoutMillis;
+
+    let delay = retry.backoffSettings.initialRetryDelayMillis;
+    let timeout = retry.backoffSettings.initialRpcTimeoutMillis;
+    let now = new Date();
+    let deadline = 0;
+
+    if (retry.backoffSettings.totalTimeoutMillis) {
+      deadline = now.getTime() + retry.backoffSettings.totalTimeoutMillis;
+    }
+    const maxRetries = retry.backoffSettings.maxRetries!;
+
+    this.timeoutAndMaxRetryCheck(
+      deadline,
+      maxRetries,
+      retry.backoffSettings.totalTimeoutMillis!
+    );
+
+    this.retries!++;
+
+    const e = GoogleError.parseGRPCStatusDetails(error);
+    let shouldRetry = this.defaultShouldRetry(e!, retry);
+    if (typeof retry.retryCodesOrShouldRetryFn! === 'function') {
+      shouldRetry = retry.retryCodesOrShouldRetryFn!(e!);
+    }
+
+    if (shouldRetry) {
+      const toSleep = Math.random() * delay;
+      setTimeout(() => {
+        now = new Date();
+        delay = Math.min(delay * delayMult, maxDelay);
+        const timeoutCal = timeout && timeoutMult ? timeout * timeoutMult : 0;
+        const rpcTimeout = maxTimeout ? maxTimeout : 0;
+        this.prevDeadline = deadline;
+        const newDeadline = deadline ? deadline - now.getTime() : 0;
+        timeout = Math.min(timeoutCal, rpcTimeout, newDeadline);
+      }, toSleep);
+    } else {
+      const newError = new GoogleError(
+        'Exception occurred in retry method that was ' +
+          'not classified as transient'
+      );
+      newError.code = Status.INVALID_ARGUMENT;
+      this.emit('error', newError);
+      this.destroy(newError);
+      return;
```

Comment:
Messing around with this right now and it swallows errors from underlying libraries, so this needs to be adjusted to make sure we surface the error to the user

Reply:
JK, I was causing a different error before it even got to the one I was expecting. It's me, hi, I'm the problem it's me

Reply:
Update, it swallows errors that aren't considered retryable. This is an issue that I'll look into

Reply:
fixed - to match the logic that we have for normal retries, this message was added as a `note` to the error and the error code and message are no longer overwritten